### PR TITLE
Iop parsing

### PR DESCRIPTION
**Pull request by ErnestaP**

### CODE REVIEW

File: dags/iop/dag_pull_ftp.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import airflow
+import common.pull_ftp as pull_ftp
+from airflow.decorators import dag, task
+from common.repository import IRepository
```
Comment (ErnestaP) [inline dags/iop/dag_pull_ftp.py]: separated local and third party impotrts

File: dags/iop/iop_process_file.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import base64
+import xml.etree.ElementTree as ET
+
+import airflow
+import requests
```
Comment (ErnestaP) [inline dags/iop/iop_process_file.py:5]: imports

File: dags/iop/parser.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+                source="front/article-meta/elocation-id",
+            ),
+            CustomExtractor(
+                destination="date_published",
+                extraction_function=self._get_date_published,
+            ),
+            TextExtractor(
+                destination="copyright_holder",
+                source="front/article-meta/permissions/copyright-holder",
+                required=False,  # copyright is required
```
Comment (ErnestaP) [inline dags/iop/parser.py:96]: there is no holder in a few files that I checked. Need to ask Anne

File: dags/iop/parser.py [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+    def _get_arxiv_eprints(self, article):
+        arxiv_eprints = []
+        arxivs_value = article.find(
+            "front/article-meta/custom-meta-group/custom-meta/meta-value"
+        ).text.lower()
+        pattern = re.compile(r"(arxiv:|v[0-9]$)", flags=re.I)
+        arxiv_eprints.append({"value": pattern.sub("", arxivs_value)})
+        return arxiv_eprints
+
+    # TODO: how to get abstract?
```
Comment (ErnestaP) [inline dags/iop/parser.py]: comment

File: dags/iop/parser.py [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+            )
+            print(string)
+            match2 = pattern2.search(string)
+            match = pattern.search(string)
+            if match2:
+                abstarct_parts += [match.group(2), match.group(4), match2.group(1)]
+            abstarct_parts += [match.group(2), match.group(4)]
+        return "".join(abstarct_parts)
+
+    # TODO: check is it correct
```
Comment (ErnestaP) [inline dags/iop/parser.py]: comment

File: dags/iop/parser.py [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+                )
+            author = {
+                "surname": surname,
+                "given_names": given_names,
+                "affiliations": affiliations,
+            }
+            authors.append(author)
+        return authors
+
+    # TODO: do I need it?
```
Comment (ErnestaP) [inline dags/iop/parser.py]: Ask Anne

File: dags/iop/parser.py [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+
+    def _get_collab():
+        pass
+
+    def _get_date_published(self, article):
+        date = article.find("front/article-meta/history/date[@date-type='published']")
+        if date:
+            day = int(date.find("day").text)
```
Comment (ErnestaP) [inline dags/iop/parser.py]: no need to create vars

File: dags/iop/parser.py [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+                url_parts = license_url.split("/")
+                clean_url_parts = list(filter(bool, url_parts))
+                version = clean_url_parts.pop()
+                licencse_type = clean_url_parts.pop()
+                licenses.append(
+                    self._construct_license(license_url, licencse_type.upper(), version)
+                )
+            elif license_text.text:
+                licenses.append(license_text.text)
+        if len(licenses):
```
Comment (ErnestaP) [inline dags/iop/parser.py]: no need it, we can just return licenses

File: tests/units/iop/test_iop_parser.py [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+
+
+def test_get_copyright_year(parsed_articles):
+    copyright_years = ["2022"]
+    for copyright_year, article in zip(copyright_years, parsed_articles):
+        assert "copyright_year" in article
+        assert article["copyright_year"] == copyright_year
+
+
+# def test_copyright_holder(parsed_articles):
```
Comment (ErnestaP) [inline tests/units/iop/test_iop_parser.py]: delete

File: dags/iop/parser.py [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+
+    def _get_date_published(self, article):
+        date = article.find("front/article-meta/history/date[@date-type='published']")
+        if date:
+            day = int(date.find("day").text)
+            month = int(date.find("month").text)
+            year = int(date.find("year").text)
+            return datetime.date(day=day, month=month, year=year).isoformat()
+        date = article.find("front/article-meta/history/date[@date-type='epub']")
+        if date:
```
Comment (ErnestaP) [inline dags/iop/parser.py]: do it more smart, please

File: dags/iop/parser.py [hunk 11]
```diff
@@ -0,0 +1,258 @@
...
+            )
+            countries = article.findall(
+                f"front/article-meta/contrib-group/aff[@id='{reffered_id.get('rid')}']/country"
+            )
+            for institution, country in zip(institutions, countries):
+                affiliations.append(
+                    {
+                        "value": ",".join([institution.text, country.text]),
+                        "country": country.text,
+                    }
```
Comment (ParthS007) [inline dags/iop/parser.py]: Might be useful to extract this part as a method and call that method to get `affiliations`.

File: dags/iop/parser.py [hunk 12]
```diff
@@ -0,0 +1,258 @@
...
+        licenses = []
+        license_texts = article.findall(
+            "front/article-meta/permissions/license/license-p/ext-link"
+        )
+        urls = article.findall("front/article-meta/permissions/license")
+        license_urls = []
+        for license_url in urls:
+            license_urls.append(
+                license_url.attrib["{http://www.w3.org/1999/xlink}href"]
+            )
```
Comment (ParthS007) [inline dags/iop/parser.py]:
```suggestion
        license_urls = [url.attrib["{http://www.w3.org/1999/xlink}href"] for url in urls]
```

File: dags/iop/iop_process_file.py [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+from airflow.decorators import dag, task
+from common.enhancer import Enhancer
+from common.enricher import Enricher
+from iop.parser import IOPParser
+from jsonschema import validate
+
+
+def iop_parse_file(**kwargs):
+    if "params" in kwargs and "file" in kwargs["params"]:
```
Comment (ParthS007) [inline dags/iop/iop_process_file.py]:
```suggestion
    if "params" in kwargs and "file" in kwargs.get("params", {}):
```

File: dags/iop/iop_process_file.py [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+        encoded_xml = kwargs["params"]["file"]
```
Comment (ParthS007) [inline dags/iop/iop_process_file.py]:
```suggestion
        encoded_xml = kwargs.get("params", {}).get("file")
```

File: dags/iop/parser.py [hunk 15]
```diff
@@ -0,0 +1,258 @@
...
+        for license_url, license_text in zip(license_urls, license_texts):
+            if license_url:
+                url_parts = license_url.split("/")
+                clean_url_parts = list(filter(bool, url_parts))
```
Comment (ParthS007) [inline dags/iop/parser.py:231]: since `url_part` is used once only.

```suggestion
                clean_url_parts = list(filter(bool,  license_url.split("/")))
```

File: dags/iop/parser.py [hunk 16]
```diff
@@ -0,0 +1,258 @@
...
+                version = clean_url_parts.pop()
+                licencse_type = clean_url_parts.pop()
```
Comment (ParthS007) [inline dags/iop/parser.py]: A typo `licencse_type` -> `license_type`

Suggestion if it is sure that last two elements are type and version respectively.
```suggestion
                version = clean_url_parts[-1]
                licencse_type = clean_url_parts[-2]
```

**Review by ParthS007 — COMMENTED** [review]
Some comments :)

File: dags/iop/repository.py [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+        )
+        for file in filenames:
+            file_parts = file.split("/")
+            last_part = file_parts[-1]
+            filename_without_extension = last_part.split(".")[0]
+            if filename_without_extension not in ret_dict.keys():
+                ret_dict[filename_without_extension] = dict()
+            ret_dict[filename_without_extension][
+                "xml" if self.is_meta(last_part) else "pdf"
+            ] = file
```
Comment (ParthS007) [inline dags/iop/repository.py]: https://github.com/cern-sis/workflows/pull/51#discussion_r993472924

File: dags/common/parsing/xml_extractors.py [hunk 18]
```diff
@@ -56,9 +56,9 @@ def __init__(
         self.required = required
 
     def extract(self, article: ET.Element):
-        value = self.extra_function(article.find(self.source).get(self.attribute))
-        if value:
-            return value
+        value = article.find(self.source)
+        if value is not None:
```
Comment (ErnestaP) [inline dags/common/parsing/xml_extractors.py:60]: value id Elementree object. If we will check with:
if value: -> will return us True
But actually, if we are checking with:
if value is None: -> will return us True

is not None/is None is necessary

File: dags/iop/parser.py [hunk 19]
```diff
@@ -0,0 +1,264 @@
...
+            doi = article.find("front/article-meta/article-id/[@pub-id-type='doi']")
+            self.logger.msg(
+                "There are unmapped article types for article %s with types %s."
+                % (doi, raw_journal_doctype)
+            )
+        return journal_doctype
+
+    def _get_related_article_doi(self, article):
+        journal_doctype = self._get_journal_doctype(article)
+        # TODO: nee to find example with related_article_doi
```
Comment (ErnestaP) [inline dags/iop/parser.py]: I wasn't able to find

File: dags/iop/parser.py (hunk 6 above)
Comment (ErnestaP) [reply dags/iop/parser.py]: Waiting for reply

File: dags/iop/iop_process_file.py [hunk 20]
```diff
@@ -1,12 +1,63 @@
...
 from airflow.decorators import dag, task
+from common.enhancer import Enhancer
+from common.enricher import Enricher
+from iop.parser import IOPParser
+from jsonschema import validate
+
+
+def iop_parse_file(**kwargs):
+    if "params" in kwargs and "file" in kwargs.get("params", {}):
+        encoded_xml = kwargs.get("params", {}).get("file")
```
Comment (drjova) [inline dags/iop/iop_process_file.py]: get_value(kawrgs, 'params.file')

File: dags/iop/parser.py [hunk 21]
```diff
@@ -0,0 +1,264 @@
...
+    def _get_journal_doctype(self, article: ET.Element):
+        raw_journal_doctype = article.find(
+            ".",
+        ).get("article-type")
+        journal_doctype = self.article_type_mapping[raw_journal_doctype]
+        if "other" in journal_doctype:
+            doi = article.find("front/article-meta/article-id/[@pub-id-type='doi']")
+            self.logger.msg(
+                "There are unmapped article types for article %s with types %s."
+                % (doi, raw_journal_doctype)
```
Comment (drjova) [inline dags/iop/parser.py]: let's use format strings instead :)

File: dags/iop/parser.py [hunk 22]
```diff
@@ -0,0 +1,264 @@
...
+                "given_names": given_names,
+                "affiliations": affiliations,
+            }
+            authors.append(author)
+        return authors
+
+    # TODO: do I need it?
+
+    def _get_collab():
+        pass
```
Comment (drjova) [inline dags/iop/parser.py]: left over?

File: dags/iop/parser.py [hunk 23]
```diff
@@ -0,0 +1,264 @@
...
+            article.find("front/article-meta/history/date[@date-type='published']")
+            or article.find("front/article-meta/history/date[@date-type='epub']")
+            or article.find("front/article-meta/history/pub-date[@pub-type='ppub']")
+            or article.find("front/article-meta/pub-date")
+        )
+        if date:
+            return self._get_date(date)
+        else:
+            # In the worst case we return today
+            return datetime.date.today().isoformat()
```
Comment (drjova) [inline dags/iop/parser.py]:
```suggestion
        if date:
            return self._get_date(date)
     
        return datetime.date.today().isoformat()
```

File: requirements.txt [hunk 24]
```diff
@@ -14,3 +14,4 @@ structlog==21.5.0
 bleach==4.1.0
 inspire-utils==3.0.25
 backoff==2.0.1
+uuid==1.30
```
Comment (drjova) [inline requirements.txt]: ?

File: tests/units/common/parsing/test_parser/input.json [hunk 25]
```diff
@@ -42,7 +42,7 @@
   ],
   "collections": ["Test Collection"],
   "control_field": ["Test control field"],
-  "free_keywords": ["Test free 1", "Test free 2"],
+  "free_keywords": ["Test free 1", "Test free  2   "],
```
Comment (drjova) [inline tests/units/common/parsing/test_parser/input.json:45]: why we need the space?

File: tests/units/iop/test_iop_trigger_files_processing.py [hunk 26]
```diff
@@ -0,0 +1,65 @@
...
+
+
+class MockLogger:
+    def msg(self, msg):
+        yield msg
+
+
+file = BytesIO(json.dumps(pseudo_article).encode("utf-8"))
+
+TRIGGERED_DAG_NAME = "iop_process_file"
```
Comment (drjova) [inline tests/units/iop/test_iop_trigger_files_processing.py]: this should be caught by pre-commit, this is not used anywhere

File: tests/units/iop/test_iop_trigger_files_processing.py [hunk 27]
```diff
@@ -0,0 +1,65 @@
...
+
+pseudo_article = {"data": "<div>Title</div>"}
+
+
+class MockLogger:
+    def msg(self, msg):
+        yield msg
+
+
+file = BytesIO(json.dumps(pseudo_article).encode("utf-8"))
```
Comment (drjova) [inline tests/units/iop/test_iop_trigger_files_processing.py]: same, it's not used anywhere

File: tests/units/iop/test_iop_trigger_files_processing.py [hunk 28]
```diff
@@ -0,0 +1,65 @@
...
+
+FIND_ALL_EXPECTED_VALUES = [
+    {"xml": "01_content.xml", "pdf": "01_content.pdf"},
+    {"xml": "02_content.xml", "pdf": "02_content.pdf"},
+]
+
+pseudo_article = {"data": "<div>Title</div>"}
+
+
+class MockLogger:
```
Comment (drjova) [inline tests/units/iop/test_iop_trigger_files_processing.py]: same is not used anywhere, could you please check this file?

File: tests/units/common/parsing/test_parser/input.json (hunk 25 above)
Comment (ErnestaP) [reply tests/units/common/parsing/test_parser/input.json:45]: I am testing if keywords are cleaned. Because these tests didn't test the cleaning part of keywords extraction at all

File: requirements.txt (hunk 24 above)
Comment (ErnestaP) [reply requirements.txt]: again.

File: dags/iop/parser.py (hunk 22 above)
Comment (ErnestaP) [reply dags/iop/parser.py]: no, I don't know do we need it. Anne said, "it might have collaborations."

Comment (drjova) [reply dags/iop/parser.py]: then we need an example and write code to parse it

File: dags/common/parsing/xml_extractors.py (hunk 18 above)
Comment (drjova) [reply dags/common/parsing/xml_extractors.py:60]: Apologies I don't understand. `.find` according to the [docs](https://docs.python.org/3/library/xml.etree.elementtree.html#xml.etree.ElementTree.Element.find) will return `None` if not found. Hence the following snippet will work.

```python
if not value:
    # do something
```

File: dags/iop/parser.py (hunk 22 above)
Comment (ErnestaP) [reply dags/iop/parser.py]: Asked Anne, to clarify about it

File: tests/integration/iop/test_pull_ftp.py [hunk 29]
```diff
@@ -0,0 +1,67 @@
...
+        repo,
+        get_logger().bind(class_name="test_logger"),
+        **{"params": {}},
+    )
+    assert len(repo.find_all()) == 11
+
+
+def test_dag_trigger_file_processing():
+    repo = IOPRepository()
+    print(repo.find_all())
```
Comment (drjova) [inline tests/integration/iop/test_pull_ftp.py]: leftover

File: dags/aps/parser.py [hunk 30]
```diff
@@ -143,10 +143,9 @@ def _get_licenses(self, x):
             licenses = []
             for right in rights:
                 url = right["url"]
-                url_parts = url.split("/")
-                clean_url_parts = list(filter(bool, url_parts))
-                version = clean_url_parts.pop()
-                license_type = clean_url_parts.pop()
+                clean_url_parts = list(filter(bool, url.split("/")))
+                version = clean_url_parts[-1]
+                license_type = clean_url_parts[-2]
```
Comment (drjova) [inline dags/aps/parser.py:148]: what if it has only one `/`? This will throw an exception. Also this is APS why are we changing it in this PR?

File: dags/iop/iop_process_file.py [hunk 31]
```diff
@@ -1,12 +1,64 @@
...
+from common.enhancer import Enhancer
+from common.enricher import Enricher
+from inspire_utils.record import get_value
+from iop.parser import IOPParser
+from jsonschema import validate
+
+
+def iop_parse_file(**kwargs):
+    if "params" in kwargs and "file" in kwargs.get("params", {}):
+        encoded_xml = get_value(kwargs, "params.file", default={})
```
Comment (drjova) [inline dags/iop/iop_process_file.py]: you don't need the if, if you use this

File: dags/iop/repository.py [hunk 32]
```diff
@@ -17,7 +17,7 @@ def __init__(self) -> None:
 
     def get_all_raw_filenames(self):
         return [
-            os.path.basename(f.key)
+            f.key.split("/")[-1]
```
Comment (drjova) [inline dags/iop/repository.py]: why this change?

File: tests/units/iop/test_iop_trigger_files_processing.py [hunk 33]
```diff
@@ -0,0 +1,53 @@
...
+    "01_content.pdf",
+    "01_content.xml",
+]
+
+FIND_ALL_EXPECTED_VALUES = [
+    {"xml": "01_content.xml", "pdf": "01_content.pdf"},
+    {"xml": "02_content.xml", "pdf": "02_content.pdf"},
+]
+
+pseudo_article = {"data": "<div>Title</div>"}
```
Comment (drjova) [inline tests/units/iop/test_iop_trigger_files_processing.py]: this is not used anywhere

File: tests/units/iop/test_iop_trigger_files_processing.py [hunk 34]
```diff
@@ -0,0 +1,53 @@
...
+    with patch("common.s3_service.boto3", autospec=True) as boto3_mock:
+        boto3_mock.resource.return_value.Bucket.return_value.objects.filter.return_value.all.return_value = [
+            S3BucketResultObj(file) for file in S3_RETURNED_VALUES
+        ]
+        yield boto3_mock
+
+
+def get_dag_runs(dag_id, states=[]):
+    dagbag = DagBag()
+    # Check DAG exists.
```
Comment (drjova) [inline tests/units/iop/test_iop_trigger_files_processing.py]: we don't need this

File: tests/units/iop/test_iop_trigger_files_processing.py [hunk 35]
```diff
@@ -0,0 +1,53 @@
+from unittest.mock import patch
+
+import pytest
+from airflow.models import DagBag
+from airflow.models.dagrun import DagRun
+
+
+class S3BucketResultObj:
+    key: str
```
Comment (drjova) [inline tests/units/iop/test_iop_trigger_files_processing.py]: why do you need this?

File: tests/units/iop/test_iop_trigger_files_processing.py [hunk 36]
```diff
@@ -0,0 +1,53 @@
...
+pseudo_article = {"data": "<div>Title</div>"}
+
+
+@pytest.fixture
+def boto3_fixture():
+    with patch("common.s3_service.boto3", autospec=True) as boto3_mock:
+        boto3_mock.resource.return_value.Bucket.return_value.objects.filter.return_value.all.return_value = [
+            S3BucketResultObj(file) for file in S3_RETURNED_VALUES
+        ]
+        yield boto3_mock
```
Comment (drjova) [inline tests/units/iop/test_iop_trigger_files_processing.py]: are we using this anywhere?

File: tests/units/iop/test_iop_process_file.py [hunk 37]
```diff
@@ -0,0 +1,40 @@
...
+from iop.iop_process_file import enhance_iop, enrich_iop, iop_validate_record
+from iop.parser import IOPParser
+from pytest import fixture
+
+
+@fixture
+def article(shared_datadir):
+    file_content = (shared_datadir / "example1.xml").read_text()
+    return ET.fromstring(file_content)
+
```
Comment (drjova) [inline tests/units/iop/test_iop_process_file.py]: in general for fixtures is better to use `yield` https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#yield-fixtures-recommended

File: tests/units/hindawi/test_hindawi_text_extractor.py [hunk 38]
```diff
@@ -0,0 +1,56 @@
...
+            articles.append(ET.fromstring(file.read()))
+    return articles
+
+
+def test_parsed_articles(hindawi_doi_extractor, articles):
+    parsed_articles = [hindawi_doi_extractor.extract(article) for article in articles]
+    assert parsed_articles == ["10.1155/2019/3465159", "10.1155/2022/5287693"]
+
+
+@fixture(scope="module")
```
Comment (drjova) [inline tests/units/hindawi/test_hindawi_text_extractor.py]: why module scope?

File: tests/units/hindawi/test_hindawi_text_extractor.py [hunk 39]
```diff
@@ -0,0 +1,56 @@
+import xml.etree.ElementTree as ET
+
+from common.parsing.xml_extractors import RequiredFieldNotFoundExtractionError
+from hindawi.xml_extractors import HindawiTextExtractor
+from pytest import fixture, raises
+
+
+@fixture(scope="module")
```
Comment (drjova) [inline tests/units/hindawi/test_hindawi_text_extractor.py]: why module scope?

File: dags/common/parsing/xml_extractors.py (hunk 18 above)
Comment (ErnestaP) [reply dags/common/parsing/xml_extractors.py:60]: Screen Shot 2022-10-27 at 14 11 26

File: dags/aps/parser.py (hunk 30 above)
Comment (ErnestaP) [reply dags/aps/parser.py:148]: I just went throw the comments, and this one was the same solution as in IOP, [Parth's comment ](https://github.com/cern-sis/workflows/pull/52#discussion_r993481829)

This error si fixed on IOP. Should I do the same for aps?

Comment (drjova) [reply dags/aps/parser.py:148]: this file is on aps :)

File: tests/units/iop/test_iop_trigger_files_processing.py (hunk 35 above)
Comment (ErnestaP) [reply tests/units/iop/test_iop_trigger_files_processing.py:9]: just to have an object, which mimics the s3 object (which has a key). For example, we need it when we are searching for all extracted articles: https://github.com/cern-sis/workflows/blob/main/dags/iop/repository.py#L55

Comment (drjova) [reply tests/units/iop/test_iop_trigger_files_processing.py]: hmm what do you mean by object? Do you need an attribute, a property, a method? 

This is an instance attribute
```python
class Hello:
     def __init__(self, key):
         self.key = key

hey = Hello('this is the key')
hey.key
> this is the key
```

Comment (drjova) [reply tests/units/iop/test_iop_trigger_files_processing.py]: Or if you need a class attribute:


```python
class Hello:
      this_is_a_class_attribute = 'Hey'

hey = Hello()
hey.this_is_a_class_attribute
> Hey
or 
Hello.this_is_a_class_attribute
> Hey
```

File: tests/units/hindawi/test_hindawi_text_extractor.py (hunk 38 above)
Comment (ErnestaP) [reply tests/units/hindawi/test_hindawi_text_extractor.py]: Because the fixture will be created once and reused, for all tests, rather than for every test. Actually, I have to do the same scope for files as well, that they won't be read per one test function

File: tests/units/hindawi/test_hindawi_text_extractor.py (hunk 39 above)
Comment (ErnestaP) [reply tests/units/hindawi/test_hindawi_text_extractor.py]: same

File: tests/units/hindawi/test_hindawi_text_extractor.py (hunk 38 above)
Comment (ErnestaP) [reply tests/units/hindawi/test_hindawi_text_extractor.py]: Sorry, I see that it is just one test. Module is not needed here

File: dags/aps/parser.py (hunk 30 above)
Comment (ErnestaP) [reply dags/aps/parser.py:148]: Yes, I understand. I can remove the fix, however, it was a good observation from Parth on IOP. And the same mistake on was done on APS. Should I revert the motdification?

File: dags/iop/parser.py (hunk 19 above)
Comment (drjova) [reply dags/iop/parser.py]: could you please remove the `#TODO` comments?

File: dags/iop/parser.py (hunk 22 above)
Comment (drjova) [reply dags/iop/parser.py]: let's completely remove this method

File: tests/units/iop/test_iop_process_file.py [hunk 40]
```diff
@@ -0,0 +1,40 @@
...
+    return enhance_iop(parsed_generic_article)
+
+
+@fixture
+def enrich_article(enhance_article):
+    return enrich_iop(enhance_article)
+
+
+def test_aps_validated_record(enrich_article):
+    iop_validate_record(enrich_article)
```
Comment (drjova) [inline tests/units/iop/test_iop_process_file.py:40]: since all these fixtures are used just for one test I would recommend to have this code inside the test

**Comment by drjova** [comment]
@ErnestaP could you please also fix the commit message according to https://scientific-information-service.web.cern.ch/SIS-TS.%20Tools%20%26%20Services/Software%20Development/Contribute/ ☺️
